client/team_role_assignment_test.go

```diff
@@ -0,0 +1,133 @@
+package client_test
+
+import (
+	. "github.com/env0/terraform-provider-env0/client"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"go.uber.org/mock/gomock"
+)
+
+var _ = Describe("TeamRoleAssignment", func() {
+	const dummyProjectAssignmentId = "dummyId"
+	const dummyProjectId = "dummyProjectId"
+	const dummyEnvironmentId = "dummyEnvironmentId"
+	const dummyOrganizationId = "dummyOrganizationId"
+	const dummyProjectRole = "Admin"
+	const dummyTeamId = "dummyTeamId"
+
+	mockTeamRoleAssignment := TeamRoleAssignmentPayload{
+		Id:     dummyProjectAssignmentId,
+		Role:   dummyProjectRole,
+		TeamId: dummyTeamId,
+	}
+
+	mockTeamRoleAssignments := []TeamRoleAssignmentPayload{mockTeamRoleAssignment}
+
+	Describe("CreateOrUpdate", func() {
+		Describe("Success", func() {
```

Comment:
please add separate tests for each scope level (org, project, env).
see if needed in the whole file.

Reply:
I added seperate test cases for the Get and Delete.
However, CreateOrUpdate behaves the same irrespective of the type. Therefore, it's a little redundant.
For completeness sake, I'll add it.